### DISCUSSION TITLE
Adding ConfigSourceInterceptor makes build fail

### DISCUSSION THREAD

**Discussion by joelmarty** [quarkiverse/quarkus-github-app#245] [Q&A] [answered]
Hi team 👋🏻 

I added a `ConfigSourceInterceptor` implementation to my github-app project and the build fails with the following error, whether I run `mvn compile`, `mvn package` or `mvn quarkus:build`.
```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.6.2.Final:build (default-cli) on project avro-inspector: Quarkus code generation phase has failed: InvocationTargetException: io.smallrye.config.ConfigSourceInterceptor: Provider com.xxx.avro.config.MyConfigSourceInterceptor not found -> [Help 1]
```

It looks like the implementation class that is registered in the `META-INF/services/io.smallrye.config.ConfigSourceInterceptor` file is not compiled prior to the code generation phase, which makes sense I suppose given that quarkus does a bunch of code generation.

How can I configure my project so that class is compiled before that phase ?

**Comment by gsmet** [comment, marked as answer]
This is going to be fixed when we upgrade to Quarkus 2.7.0.Final, which is planned for the next release.
